### PR TITLE
Document 'pdxSerializable' example

### PR DESCRIPTION
**Pull request by davebarnes97**

### CODE REVIEW

File: docs/geode-native-docs/serialization/data-serialization.html.md.erb [hunk 1]
```diff
@@ -27,3 +27,250 @@ the best combination of versatility and ease-of-use for most applications.
...
+
+The examples are located in `examples/dotnet/pdxserializable` and `examples/cpp/pdxserializable`, respectively. 
+
+## Setup
+
+PdxSerializable provides custom serialization to an individual class. Fields within an object can be serialized separately from the rest of the class.
+
+Setting up a class for PdxSerializable treatment requires some preparatory steps:
+
+- The class must inherit from the .NET `IPdxSerializable` of C++ `PdxSerializable` interface.
```
Comment (joeymcallister) [inline docs/geode-native-docs/serialization/data-serialization.html.md.erb]: I think "of" is meant to be "or" here.

File: docs/geode-native-docs/serialization/data-serialization.html.md.erb [hunk 2]
```diff
@@ -27,3 +27,250 @@ the best combination of versatility and ease-of-use for most applications.
...
+The native client release contains examples, written for .NET and C++, showing how a client application
+can register for serialiation of custom objects using the .NET `IPdxSerializable` or C++ `PdxSerializable` interface.
+
+The examples are located in `examples/dotnet/pdxserializable` and `examples/cpp/pdxserializable`, respectively. 
+
+## Setup
+
+PdxSerializable provides custom serialization to an individual class. Fields within an object can be serialized separately from the rest of the class.
+
+Setting up a class for PdxSerializable treatment requires some preparatory steps:
```
Comment (joeymcallister) [inline docs/geode-native-docs/serialization/data-serialization.html.md.erb]: I'm not sure I understand how "IPdxSerializable" and "PdxSerializable" are being used here with regard to code formatting/single ticks (i.e., sometimes they are formatted and sometimes not).

File: docs/geode-native-docs/serialization/data-serialization.html.md.erb [hunk 3]
```diff
@@ -27,3 +27,250 @@ the best combination of versatility and ease-of-use for most applications.
...
+
+- The class must inherit from the .NET `IPdxSerializable` of C++ `PdxSerializable` interface.
+- You must provide serialization instructions for objects of this class. Specifically, you must provide implementations of the 
+serialization and deserialization methods, which in .NET are named `ToData()` and `FromData()`, and in C++ arer named `toData()` and `fromData()`.
```
Comment (pdxcodemonkey) [inline docs/geode-native-docs/serialization/data-serialization.html.md.erb]: Sentence construction here feels a little awkward to me.  I would say something like "Specifically, you must implement the `ToData()` and FromData()` methods for .net, or `toData()` and `fromData()` for C++."

File: docs/geode-native-docs/serialization/data-serialization.html.md.erb [hunk 4]
```diff
@@ -35,19 +35,18 @@ The type of serialization and its implementation can be specified entirely in th
...
 
 PdxSerializable provides custom serialization to an individual class. Fields within an object can be serialized separately from the rest of the class.
 
 Setting up a class for PdxSerializable treatment requires some preparatory steps:
 
-- The class must inherit from the .NET `IPdxSerializable` of C++ `PdxSerializable` interface.
-- You must provide serialization instructions for objects of this class. Specifically, you must provide implementations of the 
-serialization and deserialization methods, which in .NET are named `ToData()` and `FromData()`, and in C++ arer named `toData()` and `fromData()`.
+- The class must inherit from the .NET IPdxSerializable interface or the C++ PdxSerializable interface.
+- You must provide serialization instructions for objects of this class. Specifically, you must implement `ToData()` and `FromData()` methods for .NET applications, and  `toData()` and `fromData()` methods in C++ applications.
```
Comment (pdxcodemonkey) [inline docs/geode-native-docs/serialization/data-serialization.html.md.erb]: Should be "you must implement _the_ ToData()..."

Comment (pdxcodemonkey) [reply docs/geode-native-docs/serialization/data-serialization.html.md.erb]: Should be "_or_ toData and..."

Comment (davebarnes97) [reply docs/geode-native-docs/serialization/data-serialization.html.md.erb]: Sure, why not...

File: docs/geode-native-docs/serialization/data-serialization.html.md.erb [hunk 5]
```diff
@@ -27,3 +27,249 @@ the best combination of versatility and ease-of-use for most applications.
...
+Once these pieces are in place, execution is simple: the main.cpp routine registers the serializable class then performs some put and get operations.
+
+## Execution
+
+Both examples perform the same sequence of operations, displaying simple log entries as they run.
+
+- To run an example, follow the instructions in the README.md file in the example directory.
+- Review the source code in the example directory to see exactly how it operates.
+
+- Begin by running a script that sets up the server-side environment by invoking `gfsh` commands to create a region.
```
Comment (igodwin) [inline docs/geode-native-docs/serialization/data-serialization.html.md.erb]: Ultimately, the startserver.[sh, ps1] scripts create a region, but the also start a locator and server. Not sure if it is worth extending this statement, though.

File: docs/geode-native-docs/serialization/data-serialization.html.md.erb [hunk 6]
```diff
@@ -27,3 +27,249 @@ the best combination of versatility and ease-of-use for most applications.
...
+```
+
+The .NET example mainline creates a cache, then uses it to register the PdxSerializable class that was created in Orders.cs:
+
+```csharp
+   var cacheFactory = new CacheFactory()
+       .Set("log-level", "none");
+   var cache = cacheFactory.Create();
+
+   cache.TypeRegistry.RegisterPdxType(Order.CreateDeserializable);
```
Comment (igodwin) [inline docs/geode-native-docs/serialization/data-serialization.html.md.erb]: The factory method CreateDeserializable() is called on the Order class in the registration. I thought we had this method noted in the "what's required for pdxserializable" section, in the past.

Comment (igodwin) [reply docs/geode-native-docs/serialization/data-serialization.html.md.erb]: It doesn't matter what the method is named, but that it returns an instance of the object. I will try to look through older documentation.

Comment (igodwin) [reply docs/geode-native-docs/serialization/data-serialization.html.md.erb]: Also, a default constructor must be provided and used by the factory method that will return the instance of the custom object.

File: docs/geode-native-docs/serialization/data-serialization.html.md.erb (hunk 3 above)
Comment (davebarnes97) [reply docs/geode-native-docs/serialization/data-serialization.html.md.erb]: Fixed.

File: docs/geode-native-docs/serialization/data-serialization.html.md.erb (hunk 5 above)
Comment (davebarnes97) [reply docs/geode-native-docs/serialization/data-serialization.html.md.erb]: Added locator and server to the list.

File: docs/geode-native-docs/serialization/data-serialization.html.md.erb (hunk 6 above)
Comment (davebarnes97) [reply docs/geode-native-docs/serialization/data-serialization.html.md.erb]: Did not add mention of default constructor, as it's specific to autoregistration on the .NET side. Did add mentions of factory class and code snippets showing its definition and use.

**Pull request merged**